stock_picking_quick/models/stock_picking.py

```diff
@@ -0,0 +1,48 @@
+# © 2019 Today Akretion
+# @author Pierrick Brun <pierrick.brun@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models, _
+
+
+class StockPicking(models.Model):
+    _name = 'stock.picking'
+    _inherit = ['stock.picking', 'product.mass.addition']
+
+    @api.multi
+    def add_product(self):
+        self.ensure_one()
+        res = self._common_action_keys()
+        res['context'].update({
+            'search_default_consumable': 1,
+            'search_default_filter_to_pick': 1,
+            'search_default_filter_for_current_location': 1,
+            'location': [self.location_id.id],
+        })
+        res['name'] = "🔙 %s" % (_('Product Variants'))
+        res['view_id'] = self.env.ref(
+            'stock_picking_quick.product_tree_view4picking').id,
+        return res
+
+    def _prepare_quick_line(self, product):
+        res = super(StockPicking, self)._prepare_quick_line(product)
+        res['location_id'] = self.location_id.id
```

Comment:
@PierrickBrun  here use write method
```
res.write({
            'location_id': self.location_id.id or False,
            'location_dest_id': self.location_dest_id.id or False
 })
```

Reply:
@nikul-serpentcs The write is done from the ProductMassAddition mixin, here: https://github.com/OCA/product-attribute/blob/b0a78c131a1da20a044414c5ba15b8ac9aff6772/base_product_mass_addition/models/product_mass_addition.py#L40